File: gaussian_unknown_mean_r/gaussian_unknown_mean_model.R
my_data_gen = function()
{
  return(list(y=rnorm(100)))
}

my_rnorm = function(n)
{
  return(rnorm(n))
}

my_dnorm = function(mu,m,s,l)
{
  return(dnorm(mu,m,s,l))
}

my_grad_log_prior = function(variable,parameters)
{
  if (variable=="mu")
  {
    stop('Not written yet.')
  }
  else
  {
    stop('Can only take the gradient with respect to "mu".')
  }
}

my_rnorm = function(n,m,s)
{
  return(rnorm(n,m,s))
}

my_dnorm = function(y,m,s,l)
{
  return(dnorm(y,m,s,l))
}

my_grad_log_llhd = function(variable,parameters,data)
{
  sigma = 1
  n = length(data$y)
  if (variable=="mu")
  {
    return(1/(sigma^2)*(-n*parameters$mu + sum(data$y)))
  }
  else
  {
    stop('Can only take the gradient with respect to "mu".')
  }
}

File: gaussian_unknown_mean_r/gaussian_unknown_mean_model.cpp
/***data,my_data_gen()***/

/***evaluate_log_prior,my_dnorm(parameters.mu,0,1,TRUE)***/

/***evaluate_gradient_log_prior,my_grad_log_prior(variable,parameters)***/

/***simulate_prior,mu=my_rnorm(1,0,1)***/

/***evaluate_log_likelihood,sum(my_dnorm(data.y,parameters.mu,1,TRUE))***/

/***evaluate_gradient_log_likelihood,my_grad_log_llhd(variable,parameters,data)***/
